src/main/scala/com/raphtory/algorithms/newer/WattsCascade.scala

```diff
@@ -0,0 +1,53 @@
+package com.raphtory.algorithms.newer
+
+import com.raphtory.core.model.algorithm.{GraphAlgorithm, GraphPerspective, Row}
+import scala.util.Random
+
+class WattsCascade(infectedSeed:Array[Long], UNIFORM_RANDOM:Int = 1, UNIFORM_SAMEVAL:Int = 2, threshold_choice:Double = 1.0, seed:Int = -1, output:String = "/tmp/wattsCascade") extends GraphAlgorithm {
+
+  val randomiser = if (seed != -1) new Random(seed) else new Random()
+
+  override def algorithm(graph: GraphPerspective): Unit = {
+    graph
+      .step({
+        vertex =>
+          if (infectedSeed.contains(vertex.ID())) {
+            vertex.setState("infected",true)
+            vertex.messageAllNeighbours(1.0)
+          }
+          else {
+            vertex.setState("infected",false)
+          }
+          if (threshold_choice == UNIFORM_RANDOM) {
+            vertex.setState("threshold", randomiser.nextFloat())
+          } else {
+            vertex.setState("threshold",threshold_choice)
+          }
+      })
+      .iterate({
+        vertex =>
+          val degree = vertex.getInEdges().size + vertex.getOutEdges().size
+          val newLabel =
+            if (degree > 0 && !vertex.getStateOrElse[Boolean]("infected", false))
+              (vertex.messageQueue[Double].sum/degree > threshold_choice)
+            else
+              vertex.getState[Boolean]("infected")
+          if (newLabel != vertex.getState[Boolean]("infected")) {
+            vertex.messageAllNeighbours(1.0)
+            vertex.setState("infected",true)
+          } else
+            vertex.voteToHalt()
+      }, 100)
```

Comment:
will need to add in the new `messagedOnly` argument as we will probably merge that first.